plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcMetadata.java

```diff
@@ -171,6 +174,21 @@ private JdbcTableHandle flushAttributesAsQuery(ConnectorSession session, JdbcTab
                 handle.getNextSyntheticColumnId());
     }
 
+    private JdbcTableHandle transformHandleIfNeeded(ConnectorSession session, JdbcTableHandle handle)
```

Comment:
I don't grasp the contract of this method (and the current logic is not correct).
I'd inline it for now.